crates/consensus/src/transaction/deposit.rs

```diff
@@ -241,6 +241,31 @@ impl Decodable for TxDeposit {
     }
 }
 
+pub trait DepositTransaction {
+    fn source_hash(&self) -> Option<B256>;
+    fn mint(&self) -> Option<u128>;
+    fn is_system_transaction(&self) -> bool;
+    fn is_deposit(&self) -> bool;
+}
```

Comment:
I'd also like to move this to a new file `src/traits.rs`

Reply:
thank @mattsse, j just update, please help me review it!